File: python/credit_check.py
import functools

def credit_check(str):
    reversed_list = list(str)
    reversed_list.reverse()

    times_two_list = times_two_e_other(reversed_list)
    return summing_results(times_two_list)


def times_two_e_other(list_nums):
    for i in range(len(list_nums)):
        if i % 2 != 0:
            if int(list_nums[i]) * 2 > 9:
                list_nums[i] = adding_up_large_nums(int(list_nums[i]) * 2)
            else:
                list_nums[i] = int(list_nums[i]) * 2
        else:
            list_nums[i] = int(list_nums[i])
    return list_nums


def adding_up_large_nums(num):
    list_num = list(str(num))
    return functools.reduce(lambda a, b: int(a) + int(b), list_num) 


def summing_results(list):
    return "The number is valid!" if functools.reduce(lambda a, b: a + b, list) % 10 == 0 else "The number is invalid!"

File: js/creditCheck.js
exports.creditCheck = function(str) {
    let reversed_arr = reversing_array(str)
    let added_two_arr = add_two_e_other_num(reversed_arr)
    return summing_results(added_two_arr)

}


function reversing_array(str) {
    let arr = str.split("")
    let reversed_arr = []

    for (let i = (arr.length) - 1; i >= 0; i--) {
        reversed_arr.push(Number(arr[i]))
    }
    return reversed_arr
}   


function add_two_e_other_num(arr) {
    let added_two = arr.map((x, i) => {
        if (i % 2 !== 0) { 
            if (x * 2 >= 10) {
                return summed_digits_over_10(x * 2)
            } else {
                return x * 2
            }
        } else {
            return x
        }
    })
    return added_two
}


function summed_digits_over_10(num) {
    let arr_num = num.toString().split("")
    let added_numbers = 0
    
    for (let i = 0; i < arr_num.length; i++) {
        added_numbers += Number(arr_num[i])
    }
    return added_numbers
}


function summing_results(arr) {
    let result = arr.reduce((accumulator, value) => accumulator + value)
    return result % 10 === 0  ? "The number is valid!" : "The number is invalid!"
}